extern/cfe/src/bmi_cfe.c

```diff
@@ -1032,7 +1032,7 @@ static int Get_value_ptr (Bmi *self, const char *name, void **dest)
 
     if (strcmp (name, "RAIN_RATE") == 0) {
         cfe_model* model = (cfe_model *)(self->data);
-        *dest = (void*)(model->forcing_data_precip_kg_per_m2 + model->current_time_step);
+        *dest = (void*)(model->forcing_data_precip_kg_per_m2 + model->current_time_step - 1);
```

Comment:
I don't fully understand why this is correct.  In the primary `run` function, rainfall input is used as the start for calculations.  In particular:

`double timestep_rainfall_input_m = model->forcing_data_precip_kg_per_m2[model->current_time_step];`

I am having trouble wrapping my head around there being an offset when outputting but not when calculating.  Can this be further explained/clarified?  

Also, if this is changed as shown, wouldn't we need to do the same thing to `SURFACE_PRESSURE` a few lines below (again, I think it just isn't being used now, but that doesn't mean it won't be later)?

Reply:
> For the first question: if you look at the aorc forcing data, the
> precip_rate (last column) is one time step ahead of APCP_surface, which is
> equivalent of rainfall. In the Ngen framework, the precip_rate is used. So
> in the original CFE model, the alignment is such that the first non-zero
> rain_rate appears at time 2015-12-02 08:00:00. The fix, though seems odd,
> aligns the rain_rate at the right time.

I don't quite understand why it's correct for the input data and calculated outputs to not apply to the same time step, but that isn't necessarily something that needs to be resolved right now.  It leaves it unclear to me, though, whether the time step shift should apply to other forcing data values, in particular to `SURFACE_PRESSURE` since right now there's code in place specifically for that one.